File: Programming-languages-and-laboratory/Python/Exercises/10_compito_java_20160704/Filiale.py
class Filiale:
    def __init__(self, nome, codice, indirizzo):
        self._nome = nome
        self._codice = codice
        self._indirizzo = indirizzo
        self._clienti_attuali = []
        self._clienti_passati = []

    def add_cliente_attuale(self, at):
        self._clienti_attuali.append(at)

    def add_cliente_passato(self, pas):
        self._clienti_passati.append(pas)

    def get_clienti_attuali(self):
        return len(self._clienti_attuali)

    def get_nome_filiale(self):
        return self._nome

    def get_clienti_totali(self):
        return len()

    def __str__(self):
        return self._nome + "\t" + str(len(self._clienti_attuali)) + "\t" + str((len(self._clienti_attuali) + len(self._clienti_passati)))

File: Programming-languages-and-laboratory/Java/Exercises/10_compito_java_2016_07_04/Filiale.java
import java.util.*;

public class Filiale {

    protected int codice;
    protected String nome, indirizzo;
    protected List<Cliente> clientiAttuali;
    protected List<Cliente> clientiPassati;

    public Filiale(int c, String n, String i) {
        codice = c;
        nome = n;
        indirizzo = i;
        clientiAttuali = new LinkedList<Cliente>();
        clientiPassati = new LinkedList<Cliente>();
    }

    public void addClAtt(Cliente c) {
        clientiAttuali.add(c);
    }

    public void addClPass(Cliente c) {
        clientiPassati.add(c);
    }

    public int nClAtt() {
        return clientiAttuali.size();
    }

    public int nClPass() {
        return clientiPassati.size();
    }

    public int getCodice() {
        return codice;
    }

    public String getNome() {
        return nome;
    }

    public String toString() {
        int na = nClAtt();
        int nt = na + nClPass();
        return nome + "\t" + na + "\t" + nt;
    }
}